This adds <sys/sysmacros.h> because the
inclusion of <sys/sysmacros.h> by <sys/types.h> is deprecated since
glibc 2.25.
diff -urp 1/src/basic/btrfs-util.c systemd-225/src/basic/btrfs-util.c
--- 1/src/basic/btrfs-util.c
+++ systemd-225/src/basic/btrfs-util.c
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <sys/vfs.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #ifdef HAVE_LINUX_BTRFS_H
 #include <linux/btrfs.h>
diff -urp 1/src/basic/terminal-util.c systemd-225/src/basic/terminal-util.c
--- 1/src/basic/terminal-util.c
+++ systemd-225/src/basic/terminal-util.c
@@ -18,6 +18,7 @@
 ***/
 
 #include <sys/ioctl.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <termios.h>
diff -urp 1/src/basic/util.c systemd-225/src/basic/util.c
--- 1/src/basic/util.c
+++ systemd-225/src/basic/util.c
@@ -30,6 +30,7 @@
 #include <sched.h>
 #include <sys/resource.h>
 #include <linux/sched.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -urp 1/src/basic/util.h systemd-225/src/basic/util.h
--- 1/src/basic/util.h
+++ systemd-225/src/basic/util.h
@@ -31,6 +31,7 @@
 #include <stdio.h>
 #include <sched.h>
 #include <limits.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
diff -urp 1/src/libsystemd/sd-device/device-private.c systemd-225/src/libsystemd/sd-device/device-private.c
--- 1/src/libsystemd/sd-device/device-private.c
+++ systemd-225/src/libsystemd/sd-device/device-private.c
@@ -19,6 +19,7 @@
 ***/
 
 #include <ctype.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <net/if.h>
 
diff -urp 1/src/libsystemd/sd-device/sd-device.c systemd-225/src/libsystemd/sd-device/sd-device.c
--- 1/src/libsystemd/sd-device/sd-device.c
+++ systemd-225/src/libsystemd/sd-device/sd-device.c
@@ -19,6 +19,7 @@
 ***/
 
 #include <ctype.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <net/if.h>
 
diff -urp 1/src/libudev/libudev-device.c systemd-225/src/libudev/libudev-device.c
--- 1/src/libudev/libudev-device.c
+++ systemd-225/src/libudev/libudev-device.c
@@ -30,6 +30,7 @@
 #include <ctype.h>
 #include <net/if.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <linux/sockios.h>
